api/src/main/java/jakarta/persistence/EntityManagerFactory.java

```diff
@@ -371,6 +372,36 @@ public interface EntityManagerFactory extends AutoCloseable {
      */
     <E> Map<String, EntityGraph<? extends E>> getNamedEntityGraphs(Class<E> entityType);
 
+    /**
+     * Register a listener for the given kind of entity lifecycle event
+     * and given entity type, returning an {@link EntityListenerRegistration}
+     * which may be used to {@linkplain EntityListenerRegistration#cancel cancel}
+     * the registration.
+     * <p>
+     * The given entity type may be an entity class or any supertype of
+     * an entity class, including {@code java.lang.Object}. The lifecycle
+     * event type is identified by the class of a callback annotation type,
+     * for example, {@link PostPersist PostPersist.class}. When a lifecycle
+     * event of the given type occurs for any entity assignable to the given
+     * entity type, the listener is invoked with the entity instance.
+     *
+     * @param entityType An entity class, or any supertype of an entity
+     *                   class
+     * @param callbackType The kind of entity callback, represented by
+     *                     the class of the callback annotation type
```

Comment:
Recommend listing the minimum set of callback annotations (the spec-defined ones) that must be supported by a Jakarta Persistence provider.